src/libcharon/sa/ikev2/tasks/ike_rekey.c

```diff
@@ -137,6 +140,19 @@ static void establish_new(private_ike_rekey_t *this)
 		{
 			lib->processor->queue_job(lib->processor, job);
 		}
+
+		const ike_super_peer_info_t* super_peer_info = this->new_sa->get_super_peer_info(this->new_sa);
```

Comment:
This would cause client builds to not compile as they don't have super_peer_info.
I suggest to move this write to an API exposed by kernel_interface. This will cover the case where kernel-super is not loaded - Meaning client builds will compile, **and the initialization of libpicomsg** can go back to kernel_super_plugin.c :)